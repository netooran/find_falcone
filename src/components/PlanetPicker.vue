<template>
  <div class="planet-picker">
    <div class="planets">
      <Planet v-for="(planet, index) in planets" :planet="planet" :key="index"/>
    </div>
  </div>
</template>

<script>
import { mapState } from "vuex";
import Planet from "@/components/Planet.vue";

export default {
  name: "PlanetPicker",
  components: { Planet },
  computed: mapState(["planets"])
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
.planets {
  display: flex;
  justify-content: space-around;
  overflow-x: scroll;
  padding: 10px;
  margin: 4vw 8% 2vw 8%;
}

.planets > * {
  margin: 5px;
}
</style>
