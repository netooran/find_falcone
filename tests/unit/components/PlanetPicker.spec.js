import Vue from "vue";
import Vuex from "vuex";
import { shallowMount } from "@vue/test-utils";
import PlanetPicker from "@/components/PlanetPicker.vue";

Vue.use(Vuex);

describe("PlanetPicker", () => {
  let wrapper, planets;

  beforeEach(() => {
    planets = [
      { name: "Toliman", distance: 437 },
      { name: "Proxima Centauri", distance: 1 }
    ];

    const store = new Vuex.Store({ state: { planets } });
    wrapper = shallowMount(PlanetPicker, { store });
  });

  it("has the correct value for planets", () => {
    expect(wrapper.vm.planets).toEqual(planets);
  });

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
