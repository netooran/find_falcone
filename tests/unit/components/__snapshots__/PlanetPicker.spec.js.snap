// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetPicker has the expected html structure 1`] = `
<div
  class="planet-picker"
>
  <div
    class="planets"
  >
    <planet-stub
      planet="[object Object]"
    />
    <planet-stub
      planet="[object Object]"
    />
  </div>
</div>
`;
